internal/services/api/post_config_secrets.go

```diff
@@ -0,0 +1,125 @@
+package api
+
+// Copyright (C) 2024 by Posit Software, PBC.
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/posit-dev/publisher/internal/config"
+	"github.com/posit-dev/publisher/internal/logging"
+	"github.com/posit-dev/publisher/internal/types"
+	"github.com/posit-dev/publisher/internal/util"
+)
+
+const (
+	secretActionAdd    = "add"
+	secretActionRemove = "remove"
+)
+
+type postConfigSecretsRequest struct {
+	Action string `json:"action"`
+	Secret string `json:"secret"`
+}
+
+func applySecretAction(cfg *config.Config, action string, secret string) error {
+	switch action {
+	case secretActionAdd:
+		// Check if the secret already exists before adding
+		for _, s := range cfg.Secrets {
+			if s == secret {
+				return nil // Secret already exists, no need to add
```

Comment:
So if duplicate, no error? Are we sure that is the desired behavior?

Reply:
I figured since Secrets should be a set, not containing duplicates, that the add should be idempotent. I can definitely change that if there is disagreement.